consensus/bor/rest.go

```diff
@@ -97,12 +100,17 @@ func (h *HeimdallClient) FetchWithRetry(rawPath string, rawQuery string) (*Respo
 	u.RawQuery = rawQuery
 
 	for {
-		res, err := h.internalFetch(u)
-		if err == nil && res != nil {
-			return res, nil
+		select {
+		case <-h.closeCh:
+			log.Info("Waiting for graceful shutdown")
+			return nil, fmt.Errorf("shutdown detected")
```

Comment:
It might be better not to use `f` if we don't have formatting. Maybe it's better to create an error variable?